port/linux/omrvmem.c

```diff
@@ -1076,61 +1091,90 @@ get_hugepages_info(struct OMRPortLibrary *portLibrary, vmem_hugepage_info_t *pag
 	return 1;
 }
 
+static BOOLEAN
+set_flags_for_mmap(int *flags)
+{
+	BOOLEAN useBackingFile = FALSE;
+#if defined(MAP_ANONYMOUS)
+	*flags |= MAP_ANONYMOUS | MAP_PRIVATE;
+#elif defined(MAP_ANON)
+	*flags |= MAP_ANON | MAP_PRIVATE;
+#else
+	useBackingFile = TRUE;
+	*flags |= MAP_PRIVATE;
+#endif
+	return useBackingFile;
+}
+
 static void *
-default_pageSize_reserve_memory(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteAmount, struct J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t pageSize, OMRMemCategory *category)
+reserve_memory_with_mmap(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteAmount, struct J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t pageSize, OMRMemCategory *category)
 {
 	/* This function is cloned in J9SourceUnixJ9VMem (omrvmem_reserve_memory).
 	 * Any changes made here may need to be reflected in that version.
 	 */
 	int fd = -1;
-	int ft = -1;
 	int flags = 0;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
-	BOOLEAN useBackingSharedFile = FALSE;
 	BOOLEAN useBackingFile = FALSE;
+	BOOLEAN useBackingSharedFile = FALSE;
+	BOOLEAN areHugePagesEnabled = FALSE;
 
-	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
+	Trc_PRT_vmem_reserve_entry(address, byteAmount);
 
-	if(mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN) {
+	if (0 != (mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN)) {
 		flags |= MAP_SHARED;
 		useBackingSharedFile = TRUE;
 	} else {
-#if defined(MAP_ANONYMOUS)
-		flags |= MAP_ANONYMOUS | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#elif defined(MAP_ANON)
-		flags |= MAP_ANON | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#else
-		useBackingFile = TRUE;
-		flags |= MAP_PRIVATE;
-#endif
+		useBackingFile = set_flags_for_mmap(&flags);
+	}
+
+#if defined(MAP_HUGETLB)
```

Comment:
Is this defined on the OpenJ9 compile platforms?
x - centOS/RHEL 6
p,z - centOS/RHEL 7

Reply:
`MAP_HUGETLB` is a `mmap` flag and exists since Linux version 2.6.32. Not sure if that answers the question

Reply:
If it's not available on the compile platforms, we need to hard code the value in OMR, otherwise the feature won't be usable in official builds.
Basically
```
#if defined(LINUX)
int map_hugeTBL = 0;
#if defined(MAP_HUGETBL)
    map_hugeTBL = MAP_HUGETBL;
#else
    /* hard coded value of MAP_HUGETBL */
    map_hugeTBL = 0x40000;
#endif
#endif

Reply:
I think we're good for MAP_HUGETLB but not MFD_HUGETLB